doc/sfp-cmis/CMIS-custom-SI-settings.md

```diff
@@ -0,0 +1,255 @@
+
+Each vendor will have per-lane SI param attribute entries applicable for the identified port + speed for the platform vendor. This value will be searched through the module key.
+
+## PORT_MEDIA_SETTINGS:  
+The entries in this block will be unique single port numbers. The control of SI attribute list generation search will reach the PORT_MEDIA_SETTINGS block only when no attribute list is generated in the GLOBAL_MEDIA_SETTINGS block.   
+
+There will be unique speed and vendor/vendor_PN entries in each identified port block.
+
+Default values can be platform defaults for multiple vendors in each section.
+
+## List of standard TX/RX SI parameters
+-  SI_PARAM_TX_INPUT_EQ 
+-  SI_PARAM_RX_OUTPUT_PRE 
+-  SI_PARAM_RX_OUTPUT_POST 
+-  SI_PARAM_RX_OUTPUT_AMP
+
+## Sample Optics SI setting file:
+```
+{
+  “TX_SETTING”: { 
+      “EQ_FIXED”: “false”, 
+      “GLOBAL_MEDIA_SETTINGS” : {}, 
+      “PORT_MEDIA_SETTINGS” : {} 
+  },
+  
+  “RX_SETTING”: { 
+      “EQ_FIXED”: “true”, 
+      “GLOBAL_MEDIA_SETTINGS” : { 
+          “1-20”: { 
+              “100G_SPEED”: { 
+                  “CREDO”: { 
+                      “CREDO_PN”: { 
+                          “SI_PARAM_RX_OUTPUT_PRE” : { 
+                              “lane0” : “5”, 
+                              “lane1” : “5”,
+                              “lane2” : “5”,
+                              “lane3” : “5”,
+                              “lane4” : “5”,
+                              “lane5” : “5”,
+                              “lane6” : “5”,
+                              “lane7” : “5”,
+                          } 
+                      } 
+                  }, 
+                  “INNOLIGHT”: { 
+                      “INNOLIGHT_PN”: { 
+                          “SI_PARAM_RX_OUTPUT_POST” : { 
+                              “lane0” : “8”,
+                              “lane1” : “8”,
+                              “lane2” : “8”,
+                              “lane3” : “8”,
+                              “lane4” : “8”,
+                              “lane5” : “8”,
+                              “lane6” : “8”, 
+                              “lane7” : “8”, 
+                          } 
+                      } 
+                  } 
+              } 
+          }, 
+          “25,28,30”: { 
+              “100G_SPEED”: { 
+                  “Default”: { 
+                      “SI_PARAM_RX_INPUT_AMP” : { 
+                          “lane0” : “7”,
+                          “lane1” : “7”,
+                          “lane2” : “7”,
+                          “lane3” : “7”,
+                          “lane4” : “7”,
+                          “lane5” : “7”,
+                          “lane6” : “7”,
+                          “lane7” : “7” 
+                      } 
+                  } 
+              } 
+          } 
+      }, 
+      “PORT_MEDIA_SETTINGS” : { 
```

Comment:
Could you please clarify the difference between per-port/range configuration in "GLOBAL_MEDIA_SETTINGS" and "PORT_MEDIA_SETTINGS" ? Both have per port configuration

Reply:
"GLOBAL_MEDIA_SETTINGS" - This setting can be applied to all ports or multiple ports. Multiple ports can be specified as a range (e.g., 1-10), a list (e.g., 1, 2, 3), or different ranges (e.g., 1-4, 8-10).

"PORT_MEDIA_SETTINGS" - This setting is specific to a single port number.

Reply:
I have checked with our HW/SI team, and they mentioned that some parameters can be port-specific as well. That's why we have PORT_MEDIA_SETTINGS.